code/game/objects/objs.dm

```diff
@@ -14,6 +14,19 @@
 	var/burn_world_time //What world time the object will burn up completely
 	var/being_shocked = 0
 
+	var/on_blueprints = FALSE //Are we visible on the station blueprints at roundstart?
+	var/force_blueprints = FALSE //forces the obj to be on the blueprints, regardless of when it was created.
+
+/obj/New()

```

Comment:
The way you implemented this is so terrible, could've ported mine but obviously you know better.

Anything which builds the initial icon after ..() in new will break here.